File: hospital/AMarotoChivite-HospitalTest/AMarotoChivite-HospitalTest-Java/Ejercicios_Intermedios_JavaMaven/src/main/java/HospitalCollections/models/Patient.java
package HospitalCollections.models;

import java.time.LocalDate;

public class Patient {

    private final String dni;
    private final String name;
    private final TypePatient typePatient;
    private final String dateNac;
    private final String dateEnter;
    private final LocalDate dateExit;

    public Patient(String dni, String name, String dateNac, String dateEnter, TypePatient typePatient) {
        this.dni = dni;
        this.name = name;
        this.dateNac = dateNac;
        this.typePatient = typePatient;
        this.dateEnter = dateEnter;
        this.dateExit = LocalDate.now();
    }


    @Override
    public String toString() {
        return "Patient{" +
                "dni='" + dni + '\'' +
                ", name='" + name + '\'' +
                ", typePatient=" + typePatient +
                ", dateNac='" + dateNac + '\'' +
                ", dateEnter='" + dateEnter + '\'' +
                ", dateExit=" + dateExit +
                '}';
    }

    public String getDni() {
        return this.dni;
    }

    public String getName() {
        return this.name;
    }

    public TypePatient getType() {
        return this.typePatient;
    }

    public String getDateEnter() {
        return this.dateEnter;
    }

    public String getDateNac() {
        return dateNac;
    }

    public LocalDate getDateExit() {
        return dateExit;
    }

    public enum TypePatient {
        NORMAL,
        URGENT
    }
}

File: hospital/AMarotoChivite-HospitalTest/AMarotoChivite-HospitalTest-Kotlin/src/main/kotlin/HospitalCollections/models/Patient.kt
package HospitalCollections.models

import java.time.LocalDate

class Patient(
    private val dni: String?,
    private val name: String?,
    private val dateNac: String?,
    private val dateEnter: String?,
    private val typePatient: TypePatient?
) {
    val dateExit: LocalDate

    init {
        dateExit = LocalDate.now()
    }

    override fun toString(): String {
        return "Patient{" +
                "dni='" + dni + '\'' +
                ", name='" + name + '\'' +
                ", typePatient=" + typePatient +
                ", dateNac='" + dateNac + '\'' +
                ", dateEnter='" + dateEnter + '\'' +
                ", dateExit=" + dateExit +
                '}'
    }

    fun getDni(): String? {
        return dni
    }

    fun getName(): String? {
        return name
    }

    val type: TypePatient?
        get() = typePatient

    fun getDateEnter(): String? {
        return dateEnter
    }

    fun getDateNac(): String? {
        return dateNac
    }

    enum class TypePatient {
        NORMAL, URGENT
    }
}